src/server/lock_manager/deadlock.rs

```diff
@@ -658,9 +659,9 @@ where
                 resp.get_deadlock_key_hash(),
             )
         }));
-        spawn_local(send.map_err(|e| error!("leader client failed"; "err" => ?e)));
+        let _ = block_on(send.map_err(|e| error!("leader client failed"; "err" => ?e)));
         // No need to log it again.
-        spawn_local(recv.map_err(|_| ()));
+        let _ = block_on(recv.map_err(|_| ()));
```

Comment:
I think this `block_on` could block forever. The deadlock detector is not written in a pure actor model.